zip-0208.rst

```diff
@@ -187,7 +215,47 @@ This default SHOULD change to BlossomPoWTargetSpacingRatio · 20 blocks after
 Blossom activation, to maintain the approximate expiry time of 50 minutes.
 
 
-TODO: check for any other number-of-block constants.
+Fingerprinting mitigation
+'''''''''''''''''''''''''
+
+A "fingerprinting attack" is a network analysis technique in which nodes are
+identified across network sessions, for example using information about which
+blocks they request or send.
+
+`zcashd` inherits from Bitcoin Core the following behaviour, described in a
+comment in `main.cpp`, intended as a fingerprinting mitigation::
+
+    // To prevent fingerprinting attacks, only send blocks outside of the active
+    // chain if they are valid, and no more than a month older (both in time, and in
+    // best equivalent proof of work) than the best header chain we know about.
+
+We make no assertion about the significance of fingerprinting for Zcash,
+and (despite the word "prevent" in the above comment) no claim about the
+effectiveness of this mitigation.
+
+In any case, to estimate the "best equivalent proof of work" of a given block
+chain (measured in units of time), we take the total work of the chain as
```

Comment:
Is this total work of the active chain, or the chain in which the block in question resides?

Reply:
The source comment says:
> To prevent fingerprinting attacks, only send blocks outside of the active chain if they are valid, and no more than a month older (both in time, and in best equivalent proof of work) than the best header chain we know about.

What that means is that we measure the total work of the chain leading up to the block in question, and also the total work of the active chain up to its tip. We take the difference (active chain minus chain-of-block-in-question), and scale it to get an approximation of the expected time for that amount of work at the current difficulty. If either that or the corresponding difference in timestamps (latest block minus block-in-question) is >= 1 month, then we don't send the block. Also we never send invalid blocks.